src/core/server/path.ts

```diff
@@ -18,34 +18,53 @@
  */
 
 import { join } from 'path';
-import { accessSync, R_OK } from 'fs';
-import { find } from 'lodash';
-import { fromRoot } from '../../utils';
+import { accessSync, constants } from 'fs';
+import { TypeOf, schema } from '@kbn/config-schema';
+import { fromRoot } from '../utils';
+
+const isString = (v: any): v is string => typeof v === 'string';
 
 const CONFIG_PATHS = [
   process.env.KIBANA_PATH_CONF && join(process.env.KIBANA_PATH_CONF, 'kibana.yml'),
-  process.env.CONFIG_PATH, //deprecated
+  process.env.CONFIG_PATH, // deprecated
```

Comment:
Shouldn't be config paths a part of `config_service`?

Reply:
`config_service` receives the array of config paths via arguments. It is initialised by the CLI handler, receiving the config via parameter (and defaulting the to the output of `getConfigPath()` (https://github.com/elastic/kibana/blob/master/src/cli/serve/serve.js#L169)

Reply:
can it be a separate file in `config` domain folder? nvm, it'd not a blocker for the current PR